packages/replayio/src/utils/createDeferred.ts

```diff
@@ -21,10 +21,15 @@ export interface Deferred<Type, Data = void> {
 
 export type Status = StatusPending | StatusRejected | StatusResolved;
 
-export function createDeferred<Type, Data = void>(
```

Comment:
The purpose for this overloaded call was so that you didn't have to pass an explicit `undefined` to deferreds with an undefined/void data type, e.g.

```ts
createDeferred<number>("debug label"); // <- This worked before but it won't now
```

Reply:
I don't feel _too strongly_ about this TBH. Was just kind of curious why the change.

Reply:
This particular call didn't work before:
> Argument of type 'string' is not assignable to parameter of type 'void'.ts(2345)

It's an ambiguous call at runtime because we can't know if the caller wants this to become the `.data` or the `.debugLabel`.

This call is still allowed using overloads:
```ts
createDeferred<number>();
```
So you don't have to use `undefined` explicitly here

Reply:
> So you don't have to use undefined explicitly here

Right, only if you wanted to pass a debug label.

But you're right that... it would be ambiguous. Sigh. I think I'm pushing my brain too hard this week. Sorry. Ignore this whole chain of feedback.